tests/components/http/test_ban.py

```diff
@@ -44,6 +58,48 @@
         assert resp.status == 403
 
 
+@pytest.mark.parametrize(
+    "remote_addr, bans, status",
+    list(zip(BANNED_IPS_WITH_SUPERVISOR, [1, 1, 0], [403, 403, 401])),
+)
+async def test_access_from_supervisor_ip(
+    remote_addr, bans, status, hass, aiohttp_client, hassio_env
+):
+    """Test accessing to server from supervisor IP."""
+    app = web.Application()
+    app["hass"] = hass
+
+    async def unauth_handler(request):
+        """Return a mock web response."""
+        raise HTTPUnauthorized
+
+    app.router.add_get("/", unauth_handler)
+    setup_bans(hass, app, 1)
+    mock_real_ip(app)(remote_addr)
+
+    with patch(
+        "homeassistant.components.http.ban.async_load_ip_bans_config", return_value=[],
+    ):
+        client = await aiohttp_client(app)
+
+    assert await async_setup_component(hass, "hassio", {"hassio": {}})
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5329694)